### PR TITLE
Make sure pgtap errors get caught

### PR DESCRIPTION
**Pull request by vincentsarago**
ref: #45

### CODE REVIEW

**Comment by vincentsarago** [comment]
I still don't get how the tests pass on GitHub while on my local machine the added tests returns 
```
{"next": null, "prev": null, "type": "FeatureCollection", "context": {"limit": 10, "matched": 0, "returned": 0}, "features": null}
```

**Comment by vincentsarago** [comment]
confirmed

the latest commit shows that the result of the pgtap command returns 
```
Checking if any tests are not ok on db ***localhost:5432/migra_to
1829
1..70
1830
ok 1 - Schema pgstac should exist
1831
ok 2 - Extension postgis should exist
1832
ok 3 - Table pgstac.migrations should exist
1833
ok 4 - Function pgstac.textarr(jsonb) should exist
1834
ok 5 - textarr returns text[] from jsonb array
1835
ok 6 - Function pgstac.estimated_count(text) should exist
1836
ok 7 - Function pgstac.textarr(jsonb) should exist
1837
ok 8 - Function pgstac.jsonb_paths(jsonb) should exist
1838
ok 9 - Function pgstac.jsonb_obj_paths(jsonb) should exist
1839
ok 10 - Function pgstac.jsonb_val_paths(jsonb) should exist
1840
ok 11 - Function pgstac.path_includes(text[], text[]) should exist
1841
ok 12 - Function pgstac.path_excludes(text[], text[]) should exist
1842
ok 13 - Function pgstac.jsonb_obj_paths_filtered(jsonb, text[], text[]) should exist
1843
ok 14 - Function pgstac.filter_jsonb(jsonb, text[], text[]) should exist
1844
ok 15 - textarr returns text[] from jsonb array
1845
ok 16 - filter_jsonb includes work
1846
ok 17 - filter_jsonb excludes work
1847
ok 18 - Function pgstac.create_cursor(text) should exist
1848
ok 19 - Function pgstac.partition_cursor(text, text) should exist
1849
ok 20 - Function pgstac.stac_geom(jsonb) should exist
1850
ok 21 - Function pgstac.stac_datetime(jsonb) should exist
1851
ok 22 - Function pgstac.stac_end_datetime(jsonb) should exist
1852
ok 23 - Function pgstac.stac_daterange(jsonb) should exist
1853
ok 24 - Function pgstac.properties_idx(jsonb) should exist
1854
ok 25 - Table pgstac.collections should exist
1855
ok 26 - collections has primary key
1856
ok 27 - Function pgstac.create_collection(jsonb) should exist
1857
ok 28 - Function pgstac.update_collection(jsonb) should exist
1858
ok 29 - Function pgstac.upsert_collection(jsonb) should exist
1859
ok 30 - Function pgstac.get_collection(text) should exist
1860
ok 31 - Function pgstac.delete_collection(text) should exist
1861
ok 32
1862
ok 33 - Table pgstac.items should exist
1863
ok 34 - Should have an index on pgstac.items(geometry)
1864
ok 35 - Should have an index on pgstac.items(collection_id)
1865
ok 36 - Table pgstac.items should be partitioned
1866
ok 37 - Function pgstac.get_item(text) should exist
1867
ok 38 - Function pgstac.delete_item(text) should exist
1868
ok 39 - Function pgstac.create_item(jsonb) should exist
1869
ok 40 - Function pgstac.update_item(jsonb) should exist
1870
ok 41 - Function pgstac.upsert_item(jsonb) should exist
1871
ok 42
1872
ok 43 - all_items_partitions view exists
1873
ok 44 - items_partitions view exists
1874
ok 45 - Function pgstac.collection_bbox(text) should exist
1875
ok 46 - Function pgstac.collection_temporal_extent(text) should exist
1876
ok 47
1877
ok 48 - Function pgstac.parse_dtrange(jsonb) should exist
1878
ok 49 - daterange passed as array range
1879
ok 50 - date range passed as string range
1880
ok 51 - Function pgstac.bbox_geom(jsonb) should exist
1881
ok 52 - 2d bbox
1882
ok 53 - 3d bbox
1883
ok 54 - Function pgstac.add_filters_to_cql(jsonb) should exist
1884
ok 55 - Test that id gets added to cql filter when cql filter does not exist
1885
ok 56 - Test that id gets added to cql filter when cql filter does exist
1886
ok 57 - Test that collections gets added to cql filter when cql filter does not exist
1887
ok 58 - Test that collection are not added to cql filter
1888
ok 59 - Function pgstac.cql_and_append(jsonb, jsonb) should exist
1889
ok 60 - Function pgstac.query_to_cqlfilter(jsonb) should exist
1890
ok 61 - Test that query_to_cqlfilter appropriately converts old style query items to cql filters
1891
ok 62 - Function pgstac.sort_sqlorderby(jsonb, boolean) should exist
1892
ok 63 - Test creation of sort sql
1893
ok 64 - Test creation of reverse sort sql
1894
ok 65 - Test basic search with fields and sort extension
1895
not ok 66 - Test basic search with fields and sort extension and next token
1896
# Failed test 66: "Test basic search with fields and sort extension and next token"
1897
#     Results differ beginning at row 1:
1898
#         have: ("{""next"": ""pgstac-test-item-0020"", ""prev"": ""pgstac-test-item-0007"", ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 10}, ""features"": [{""id"": ""pgstac-test-item-0007"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0008"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 64}}, {""id"": ""pgstac-test-item-0009"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 61}}, {""id"": ""pgstac-test-item-0014"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 17}}, {""id"": ""pgstac-test-item-0015"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 54}}, {""id"": ""pgstac-test-item-0016"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 13}}, {""id"": ""pgstac-test-item-0017"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0018"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 29}}, {""id"": ""pgstac-test-item-0019"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 52}}, {""id"": ""pgstac-test-item-0020"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 39}}]}")
1899
#         want: ("{""next"": ""pgstac-test-item-0020"", ""prev"": ""pgstac-test-item-0011"", ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 10}, ""features"": [{""id"": ""pgstac-test-item-0011"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 41}}, {""id"": ""pgstac-test-item-0012"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 4}}, {""id"": ""pgstac-test-item-0013"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 2}}, {""id"": ""pgstac-test-item-0014"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 17}}, {""id"": ""pgstac-test-item-0015"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 54}}, {""id"": ""pgstac-test-item-0016"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 13}}, {""id"": ""pgstac-test-item-0017"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0018"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 29}}, {""id"": ""pgstac-test-item-0019"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 52}}, {""id"": ""pgstac-test-item-0020"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 39}}]}")
1900
not ok 67 - Test basic search with fields and sort extension and prev token
1901
# Failed test 67: "Test basic search with fields and sort extension and prev token"
1902
#     Results differ beginning at row 1:
1903
#         have: ("{""next"": ""pgstac-test-item-0013"", ""prev"": ""pgstac-test-item-0001"", ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 8}, ""features"": [{""id"": ""pgstac-test-item-0001"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 89}}, {""id"": ""pgstac-test-item-0002"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 33}}, {""id"": ""pgstac-test-item-0003"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 28}}, {""id"": ""pgstac-test-item-0004"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 23}}, {""id"": ""pgstac-test-item-0005"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 3}}, {""id"": ""pgstac-test-item-0006"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 100}}, {""id"": ""pgstac-test-item-0012"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 4}}, {""id"": ""pgstac-test-item-0013"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 2}}]}")
1904
#         want: ("{""next"": ""pgstac-test-item-0010"", ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 10}, ""features"": [{""id"": ""pgstac-test-item-0001"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 89}}, {""id"": ""pgstac-test-item-0002"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 33}}, {""id"": ""pgstac-test-item-0003"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 28}}, {""id"": ""pgstac-test-item-0004"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 23}}, {""id"": ""pgstac-test-item-0005"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 3}}, {""id"": ""pgstac-test-item-0006"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 100}}, {""id"": ""pgstac-test-item-0007"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0008"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 64}}, {""id"": ""pgstac-test-item-0009"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 61}}, {""id"": ""pgstac-test-item-0010"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 31}}]}")
1905
ok 68 - Test datetime search with datetime as / separated string
1906
ok 69 - Test datetime search with datetime as array
1907
ok 70 - Test datetime as an anyinteracts filter
1908
ok 71 - Test equality as a filter on a numeric field
1909
ok 72 - Test lt as a filter on a numeric field with order by
1910
not ok 73 - Test collections search with unknow collection
1911
# Failed test 73: "Test collections search with unknow collection"
1912
#     Results differ beginning at row 1:
1913
#         have: ("{""next"": ""pgstac-test-item-0003"", ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 1, ""matched"": 100, ""returned"": 1}, ""features"": [{""id"": ""pgstac-test-item-0003""}]}")
1914
#         want: ("{""next"": ""20200307aC0870130w361200"", ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 1, ""matched"": 100, ""returned"": 1}, ""features"": [{""id"": ""20200307aC0870130w361200""}]}")
1915
ok 74 - Test collections search with unknow collection
1916
not ok 75 - Test collections search return empty feature not null
1917
# Failed test 75: "Test collections search return empty feature not null"
1918
#     Results differ beginning at row 1:
1919
#         have: ("{""next"": null, ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 0, ""returned"": 0}, ""features"": null}")
1920
#         want: ("{""next"": null, ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 0, ""returned"": 0}, ""features"": []}")
1921
# Looks like you planned 70 tests but ran 75
``` 
but the `test` https://github.com/stac-utils/pgstac/blob/d1f4df4d84ae5069877d507cc58ace464a992507/scripts/bin/migra_funcs#L48-L54 does not work

File: scripts/bin/migra_funcs [hunk 1]
```diff
@@ -45,9 +45,9 @@ export -f pgwait
 function pgtap(){
     TESTOUTPUT=$(psql -X -f $BASEDIR/test/pgtap.sql $1)
     echo "Checking if any tests are not ok on db $1"
-    if [[ $(echo "$TESTOUTPUT" | grep '^not') == 0 ]]; then
+    if [[ $(echo "$TESTOUTPUT" | grep -e '^not') ]]; then
         echo "PGTap tests failed."
-        echo "$TESTOUTPUT" | grep '^not'
+        echo "$TESTOUTPUT"
```
Comment (vincentsarago) [inline scripts/bin/migra_funcs:50]: Let's print all the output from pgtap to help debug

File: test/pgtap/004_search.sql [hunk 2]
```diff
@@ -170,6 +170,14 @@ SELECT results_eq($$
     'Test collections search with unknow collection'
 );
 
+SELECT results_eq($$
+    select s from search('{"collections":["something"],"fields":{"include":["id"]}}') s;
+    $$,$$
+    select '{"next": null, "prev": null, "type": "FeatureCollection", "context": {"limit": 10, "matched": 0, "returned": 0}, "features": []}'::jsonb
+    $$,
+    'Test collections search return empty feature not null'
+);
```
Comment (vincentsarago) [inline test/pgtap/004_search.sql:179]: add tests to prove #45

**Comment by vincentsarago** [comment]
🥳 

we now have 3 tests failing 

```
not ok 66 - Test basic search with fields and sort extension and next token
# Failed test 66: "Test basic search with fields and sort extension and next token"
#     Results differ beginning at row 1:
#         have: ("{""next"": ""pgstac-test-item-0020"", ""prev"": ""pgstac-test-item-0007"", ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 10}, ""features"": [{""id"": ""pgstac-test-item-0007"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0008"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 64}}, {""id"": ""pgstac-test-item-0009"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 61}}, {""id"": ""pgstac-test-item-0014"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 17}}, {""id"": ""pgstac-test-item-0015"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 54}}, {""id"": ""pgstac-test-item-0016"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 13}}, {""id"": ""pgstac-test-item-0017"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0018"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 29}}, {""id"": ""pgstac-test-item-0019"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 52}}, {""id"": ""pgstac-test-item-0020"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 39}}]}")
#         want: ("{""next"": ""pgstac-test-item-0020"", ""prev"": ""pgstac-test-item-0011"", ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 10}, ""features"": [{""id"": ""pgstac-test-item-0011"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 41}}, {""id"": ""pgstac-test-item-0012"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 4}}, {""id"": ""pgstac-test-item-0013"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 2}}, {""id"": ""pgstac-test-item-0014"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 17}}, {""id"": ""pgstac-test-item-0015"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 54}}, {""id"": ""pgstac-test-item-0016"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 13}}, {""id"": ""pgstac-test-item-0017"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0018"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 29}}, {""id"": ""pgstac-test-item-0019"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 52}}, {""id"": ""pgstac-test-item-0020"", ""properties"": {""datetime"": ""2011-08-16T00:00:00Z"", ""eo:cloud_cover"": 39}}]}")

not ok 67 - Test basic search with fields and sort extension and prev token
# Failed test 67: "Test basic search with fields and sort extension and prev token"
#     Results differ beginning at row 1:
#         have: ("{""next"": ""pgstac-test-item-0013"", ""prev"": ""pgstac-test-item-0001"", ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 8}, ""features"": [{""id"": ""pgstac-test-item-0001"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 89}}, {""id"": ""pgstac-test-item-0002"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 33}}, {""id"": ""pgstac-test-item-0003"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 28}}, {""id"": ""pgstac-test-item-0004"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 23}}, {""id"": ""pgstac-test-item-0005"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 3}}, {""id"": ""pgstac-test-item-0006"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 100}}, {""id"": ""pgstac-test-item-0012"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 4}}, {""id"": ""pgstac-test-item-0013"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 2}}]}")
#         want: ("{""next"": ""pgstac-test-item-0010"", ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 100, ""returned"": 10}, ""features"": [{""id"": ""pgstac-test-item-0001"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 89}}, {""id"": ""pgstac-test-item-0002"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 33}}, {""id"": ""pgstac-test-item-0003"", ""properties"": {""datetime"": ""2011-08-25T00:00:00Z"", ""eo:cloud_cover"": 28}}, {""id"": ""pgstac-test-item-0004"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 23}}, {""id"": ""pgstac-test-item-0005"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 3}}, {""id"": ""pgstac-test-item-0006"", ""properties"": {""datetime"": ""2011-08-24T00:00:00Z"", ""eo:cloud_cover"": 100}}, {""id"": ""pgstac-test-item-0007"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 59}}, {""id"": ""pgstac-test-item-0008"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 64}}, {""id"": ""pgstac-test-item-0009"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 61}}, {""id"": ""pgstac-test-item-0010"", ""properties"": {""datetime"": ""2011-08-17T00:00:00Z"", ""eo:cloud_cover"": 31}}]}")

not ok 75 - Test collections search return empty feature not null
# Failed test 75: "Test collections search return empty feature not null"
#     Results differ beginning at row 1:
#         have: ("{""next"": null, ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 0, ""returned"": 0}, ""features"": null}")
#         want: ("{""next"": null, ""prev"": null, ""type"": ""FeatureCollection"", ""context"": {""limit"": 10, ""matched"": 0, ""returned"": 0}, ""features"": []}")
``` 
test 75, shows the behaviour of #45. The two others will need deeper looks

File: test/pgtap/004_search.sql [hunk 3]
```diff
@@ -157,9 +157,9 @@ SELECT results_eq($$
 SELECT results_eq($$
     select s from search('{"collections":["pgstac-test-collection"],"fields":{"include":["id"]}, "limit": 1}') s;
     $$,$$
-    select '{"next": "20200307aC0870130w361200", "prev": null, "type": "FeatureCollection", "context": {"limit": 1, "matched": 100, "returned": 1}, "features": [{"id": "20200307aC0870130w361200"}]}'::jsonb
+    select '{"next": "pgstac-test-item-0003", "prev": null, "type": "FeatureCollection", "context": {"limit": 1, "matched": 100, "returned": 1}, "features": [{"id": "pgstac-test-item-0003"}]}'::jsonb
     $$,
-    'Test collections search with unknow collection'
+    'Test collections search with collection'
```
Comment (vincentsarago) [inline test/pgtap/004_search.sql:162]:
```suggestion
    'Test collections search'
```

File: sql/004_search.sql [hunk 4]
```diff
@@ -845,7 +843,7 @@ context := jsonb_strip_nulls(jsonb_build_object(
 
 collection := jsonb_build_object(
     'type', 'FeatureCollection',
-    'features', out_records,
+    'features', coalesce(out_records, '[]'::jsonb),
```
Comment (vincentsarago) [inline sql/004_search.sql:846]: simple fix to make sure `features` is at least set to `[]`.

I couldn't track down where in https://github.com/stac-utils/pgstac/pull/46/files#diff-3c3e93de12c078e7f46e264188fcc2ba82eefd77be73795f964ad132d89af285R835 the transformation from `[]` to `null` was 🤷

Comment (vincentsarago) [reply sql/004_search.sql:846]: 😬 tests still fails 🤷‍♂️ 
😭 but it works locally 😭

**Comment by bitner** [comment]
@vincentsarago I think one of your tests was failing due to not having regenerated the migrations files (/scripts/stageversion). The other two were failing because there was an additional place where the error I fixed before I left with "sort" vs "sortby" was still wrong. When looking for this, I also fixed and added tests for a bug where I had used "id" rather than "ids".

**Review by bitner — APPROVED** [review]

**Pull request merged**